EIPS/eip-7336.md

```diff
@@ -0,0 +1,104 @@
+---
+eip: 7336
+title: Interactive across multiple registries
+description: An Universal Domain Namespace(UDoN) interface allowing for interactive capabilities across multiple registries like ENS
+author: Mengshi Zhang (@MengshiZhang), Zihao Chen (@zihaoccc)
+discussions-to: https://ethereum-magicians.org/t/erc7336-a-interface-allowing-for-interactive-capabilities-across-multiple-registries-like-ens/15069
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-14
+requires: 165
+---
+
+## Abstract
+
+The objective of this EIP is to propose the **Universal Domain Namespace(UDoN)**, which aims to facilitate onchain registry entries. Each entry in this structure adopts a mapping type, allowing for interactive capabilities across multiple registries like [ENS](./eip-137.md) and [NFT accounts](./eip-6551.md).
+
+## Motivation
+
+The necessity for a unified method of data retrieval across various registries is apparent. A structured methodology that allows for seamless reading of data from one registry to another, effectively linking chains of information, is thus proposed. This proposal seeks to standardize an interface that not only supports interaction but also promotes integration with multi-protocol registries. With the utilization of such an interface, a single deployed contract can retrieve records from different combinations of registries, such as ENS, the registry for royalty payments of NFTs, the registry for smart contract accounts owned by NFT tokens, and other registries as defined by varying protocols.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**The Universal Domain Namespace(UDoN)** is a singular contract that enables the retrieval of data from multiple registries, each defined under different protocols within its own namespace. These registries employ distinct methods for resolving the key-value pairs.
```

Comment:
EIPs are generally used to standardize ideas that require interoperability between multiple vendors. If there is a single UDoN registry, it may be more suited to be developed as a service or library as opposed to an EIP.

That's not to say you cannot propose UDoN here, but I would like to see some text in your Motivation that explains why this needs an EIP over some documentation on, for example, a GitHub Pages site.